### PR TITLE
AppShow via the controller only returns an app

### LINKED ISSUE #633: Deploying to the rack name gives confusing error
If you run `convox deploy -a $rackname` everything looks like it tries to work and then you get a very confusing error message about an S3 bucket. We should validate the app name parameter here and return a helpful error message.

/cc @kmonkeyjam

### PR DESCRIPTION
**Pull request by MiguelMoll**
If asked for a rack via the AppShow method an error will be returned. Displays a nice error which fixes #633.

### CODE REVIEW

**Comment by MiguelMoll** [comment]
Before (`dev` being the currect rack):
``` bash
⟩ convox deploy -a dev
Deploying misc
Creating tarball... OK
Uploading... ERROR: exit status 1
```

After:
``` bash
⟩ convox deploy -a dev
Deploying dev
ERROR: rack dev is not an app
```

Using a rack that's not the current one:
``` bash
⟩ convox deploy -a misc
Deploying misc
ERROR: invalid rack: misc
```

**Comment by MiguelMoll** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;VQIadVqD&#x2F;254-prevent-the-deployment-of-rack"> Prevent the deployment of rack</a>

File: api/controllers/apps.go [hunk 1]
```diff
@@ -30,8 +29,11 @@ func AppList(rw http.ResponseWriter, r *http.Request) *httperr.Error {
 func AppShow(rw http.ResponseWriter, r *http.Request) *httperr.Error {
 	app := mux.Vars(r)["app"]
 
-	a, err := models.GetApp(mux.Vars(r)["app"])
+	if app == os.Getenv("RACK") {
+		return httperr.Errorf(404, "rack %s is not an app", app)
```
Comment (nzoschke) [inline api/controllers/apps.go]: Nice, I like the message and error code here.

**Comment by nzoschke** [comment]
👍 looks good to me. Builds / deploys will fail on their app/status lookup, but the logs endpoint will still work.

We'll want to continue to tighten the validation of app and rack names -> stacks, but the most bang for the buck will be doing that as part of another provider interface refactor iteration.

**Pull request closed without merging**